socket/src/commonTest/kotlin/mqtt/socket/ServerProcessTest.kt

```diff
@@ -0,0 +1,104 @@
+    public var name: String = ""
+    public var clientResponse: String = ""
+
+    @ExperimentalUnsignedTypes
+    val limits = object : BufferMemoryLimit {
+        override fun isTooLargeForMemory(size: UInt) = size > 1_000u
+    }
+
+    private suspend fun ConnectProcess () {
+
+    }
+
+    @ExperimentalTime
+    private suspend fun ConnectDisProcess () {
+        val serverReadBuffer = allocateNewBuffer(10.toUInt(), limits)
+
+        try {
+            assertTrue(socket.isOpen(), "socket to client is not open")
+            socket.read(serverReadBuffer, timeout)
+        } catch (e: Exception) {
+            assertEquals("remote returned -1 bytes", e.message, "Unknown exception recevied")
+            socket.close()
+        }
+    }
+
+    @ExperimentalTime
+    private suspend fun UShortProcess() {
+        val serverReadBuffer = allocateNewBuffer(10.toUInt(), limits)
+        val serverWriteBuffer = allocateNewBuffer(10.toUInt(), limits)
+        val recvData: UShort = 4.toUShort()
+        val sendData: UInt = UInt.MAX_VALUE
+
+        try {
+            assertTrue(socket.isOpen(), "socket to client is not open")
+            assertEquals(2, socket.read(serverReadBuffer, timeout), "server received invalid data")
+            assertEquals(recvData, serverReadBuffer.readUnsignedShort(), "server received invalid data")
+            serverWriteBuffer.write(sendData)
+            assertEquals(4, socket.write(serverWriteBuffer, timeout), "server send data size not correct")
+            socket.close()
+        } catch (e: Exception) {
+            println("ServerProcessUshort.exception: ${e.message}, $e")
+            assertEquals("Failure...", e.message)
+        }
+    }
+
+    @ExperimentalTime
+    @ExperimentalUnsignedTypes
+    private suspend fun MqttStringProcess() {
+        val rbuffer = allocateNewBuffer(100.toUInt(), limits)
+        val wbuffer = allocateNewBuffer(100.toUInt(), limits)
+
+        try {
+            assertTrue(socket.isOpen(), "Client socket is not open")
+
+            //assertEquals(socket.read(rbuffer, timeout), clientResponse.length + 3, "message read length not correct")
+
+            val ret: Int = socket.read(rbuffer, timeout)
+
+            var str: String = rbuffer.readMqttUtf8StringNotValidated().toString()
+            assertEquals(ret, str.length + 2, "message read length not correct")
+ //           println("==> $ret,, $str, $rbuffer")
+            assertEquals(clientResponse, str.substring(0, clientResponse.length), "Received message is not correct.")
+
+            wbuffer.writeUtf8String(str + ":" + name)
+
+            assertEquals(socket.write(wbuffer, timeout), name.length + str.length + 3, "write message length not correct")
+        } catch (e: Exception) {
+            println("TestServerProcess.serverSideProcess.exception: $e, ${e.message}")
+        }
+    }
+
+    override suspend fun newInstance(): ServerProcess {
+        val x: ServerProcessTest = ServerProcessTest(action)
```

Comment:
can we come up with a better variable name?
`val process = ServerProcessTest(action)`?